dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,293 @@
+
+            # our objective
+            np.fill_diagonal(correlations, -correlations[:, -1] * alpha)
+
+            # Note: the full symmetric matrix (with both upper- and lower-diagonal
+            # entries for each correlation coefficient) is retained for consistency with
+            # the original formulation from Milne et al.
+            it = np.nditer(correlations[:-1, :-1], flags=['multi_index'], op_flags=[['readonly']])
+            cqm.set_objective((*it.multi_index, x) for x in it)
+
+        return cqm
+
+    def fit(
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: typing.Optional[float] = None,
+        num_features: typing.Optional[int] = None,
+        time_limit: typing.Optional[float] = None,
+    ) -> SelectFromQuadraticModel:
+        """Select the features to keep.
+
+        Args:
+            X:
+                Feature vectors formatted as a numerical 2D array-like.
+            y:
+                Class labels formatted as a numerical 1D array-like.
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.  `alpha=1` places all weight on
+                relevance and selects all features, whereas `alpha=0` places all
+                weight on redundancy and selects no features.
+                Defaults to the value provided to the constructor.
+            num_features:
+                The number of features to select.
+                Defaults to the value provided to the constructor.
+            time_limit:
+                The time limit for the run on the hybrid solver.
+                Defaults to the value provided to the constructor.
+
+        Returns:
+            This instance of `SelectFromQuadraticModel`.
+        """
+        X = np.atleast_2d(np.asarray(X))
+        if X.ndim != 2:
+            raise ValueError("X must be a 2-dimensional array-like")
+
+        # y is checked by the correlation method function
+
+        if alpha is None:
+            alpha = self._alpha
+        # alpha is checked by the correlation method function
+
+        if num_features is None:
+            num_features = self._num_features
+        # num_features is checked by the correlation method function
+
+        # time_limit is checked by the LeapHybridCQMSampelr
+
+        # if we already have fewer features than requested, just return
+        if num_features >= X.shape[1]:
+            self._mask = np.ones(X.shape[1], dtype=bool)
+            return self
+
+        if self._method == "correlation":
+            cqm = self.correlation_cqm(X, y, num_features=num_features, alpha=alpha)
+        # elif self._method == "mutual information":
+        #     cqm = self.mutual_information_cqm(X, y, num_features=num_features)
+        else:
+            raise ValueError(f"only methods {self.acceptable_methods} are implemented")
```

Comment:
this gets checked at instantiation and raises an error there, so it's prob redundant. I was only checking here because of MI, but that's commented out now.

Reply:
Here it's more to signal design intention to future developers than actual safeguard. So catches the case that someone naively adds something to `.ACCEPTED_METHODS` without updating the code.